stages/org.osbuild.rpm

```diff
@@ -303,13 +303,17 @@ def main(tree, inputs, options):
 
     for filename, data in packages.items():
         if data.get("rpm.check_gpg"):
-            subprocess.run([
-                "rpmkeys",
-                *rpm_args,
-                "--root", tree,
-                "--checksig",
-                filename
-            ], cwd=pkgpath, stdout=subprocess.DEVNULL, check=True)
+            try:
+                subprocess.run([
+                    "rpmkeys",
+                    *rpm_args,
+                    "--root", tree,
+                    "--checksig",
+                    filename
+                ], cwd=pkgpath, stdout=subprocess.DEVNULL, check=True)
+            except Exception:
+                print(f"Signature check failed on {filename}, lookup package name in manifest.")
```

Comment:
Would it be better if print output to stderr ? It seems rarely done in the stages though.

Reply:
I would consider this to be a minor thing, because we are not consistent in this:

```
thozza@fedora ~/devel/osbuild/stages (main $=)
$ grep -r 'print('
org.osbuild.users:                print(f"Error: can't set uid of existing user '{name}'")
org.osbuild.copy:        print(f"copying '{src}' -> '{dst}'")
org.osbuild.dracut:        print(f"Building initramfs for {kver}", file=sys.stderr)
org.osbuild.error:    print(f"Error stage will now return error: {errno}")
org.osbuild.grub2.legacy:            print(data)
org.osbuild.grub2.legacy:        print(data)
org.osbuild.timezone:        print("/etc/localtime already exists. Replacing.")
org.osbuild.fstab:        print(f"ostree support active: {root}")
org.osbuild.grub2.iso:    print(f"kernel dir at {kdir}")
org.osbuild.hostname:        print("/etc/hostname already exists. Replacing.")
org.osbuild.grub2:        print("hybrid boot or unified grub config enabled. "
org.osbuild.grub2:        print(data)
org.osbuild.implantisomd5:    print(cmd)
org.osbuild.keymap:        print("/etc/vconsole.conf already exists. Replacing.")
org.osbuild.ostree.config:    print("ostree " + " ".join(args), file=sys.stderr)
org.osbuild.locale:        print("/etc/locale.conf already exists. Replacing.")
org.osbuild.ostree.deploy:    print("ostree " + " ".join(args), file=sys.stderr)
org.osbuild.lorax-script:    print(f"running script: {os.path.dirname(filename)}")
org.osbuild.ostree.init-fs:    print("ostree " + " ".join(args), file=sys.stderr)
org.osbuild.noop:    print("Not doing anything with these options:", json.dumps(options))
org.osbuild.noop:        print(f"Not doing anything with Input '{name}'", json.dumps(desc))
org.osbuild.ostree.os-init:    print("ostree " + " ".join(args), file=sys.stderr)
org.osbuild.oci-archive:    print(f"blobs: +{mtype} ({size}, {digest})")
org.osbuild.oci-archive:    print("creating layer")
org.osbuild.oci-archive:    print("compressing layer")
org.osbuild.oci-archive:    print(config)
org.osbuild.oci-archive:    print("writing index")
org.osbuild.oci-archive:        print("creating final archive")
org.osbuild.ostree:    print("ostree " + " ".join(args), file=sys.stderr)
org.osbuild.ostree.remotes:    print("ostree " + " ".join(args), file=sys.stderr)
org.osbuild.ostree.preptree:    print(f"ostree: machineid-compat: {machineid_compat}")
org.osbuild.ostree.preptree:        print("Moving tree to temporary root")
org.osbuild.ostree.preptree:        print("Initializing new root filesystem")
org.osbuild.ostree.preptree:        print("Moving data back from temporary root")
org.osbuild.ostree.pull:    print("ostree " + " ".join(args), file=sys.stderr)
org.osbuild.qemu:        print(f"qemu-img coroutines: {coroutines}")
org.osbuild.resolv-conf:        print(f"Replacing existing file: /{filepath}")
org.osbuild.rhsm:            print(f"Error: {plugin} configuration file '{plugin_conf_path}' does not exist.")
org.osbuild.rhsm:                print(f"Error: unknown property {option} specified for {plugin} plugin.")
org.osbuild.rhsm:        print(f"Error: RHSM configuration file '{rhsm_config_path}' does not exist.")
org.osbuild.yum.config:            print(f"Warning: {plugin} configuration file '{plugin_conf_path}' does not exist, will create it.")
org.osbuild.yum.config:                print(f"Error: unknown property {option} specified for {plugin} plugin.")
org.osbuild.yum.config:            print(f"Warning: YUM configuration file '{yum_config_path}' does not exist, will create it.")
org.osbuild.rpm:        print(f"/etc/kernel is owned by {owner}")
org.osbuild.rpm:        print(f"dbpath set to '{dbpath}'")
org.osbuild.rpm:        print("imported gpg key")
org.osbuild.rpm:        print(f"using '{macro}'")
org.osbuild.rpm:        print(f"imported gpg keys from '{key}'")
org.osbuild.rpm:        print("deleting the fake machine id")
org.osbuild.bootiso.mono:            print("copying tree")
org.osbuild.bootiso.mono:            print("done")
org.osbuild.bootiso.mono:    print(f"kernel dir at {kdir}")
org.osbuild.bootiso.mono:    print(f"Estimates efiboot size to be {size}")
org.osbuild.rpm-ostree:    print(f"ostree: machineid-compat: {machineid_compat}")
org.osbuild.grub2.inst:    print(f"sector start param: {dest}")
org.osbuild.grub2.inst:    print(f"grub core size is {core_size}")
org.osbuild.grub2.inst:    print(f"wiring core to ({location}, {location_bytes})")
org.osbuild.grub2.inst:    print(f"prefix: {prefix}")
org.osbuild.dnf.config:        print(f"Warning: DNF configuration file '{dnf_config_path}' does not exist, will create it.")
org.osbuild.sfdisk:        print(command)
org.osbuild.kickstart:    print(f"created kickstarted at: {path}\n")
org.osbuild.kickstart:        print(f.read())
org.osbuild.lvm2.create:    print(f"LVM2: using vg name '{vg_name}'")
org.osbuild.xorrisofs:    print(cmd)
org.osbuild.pacman:        print("deleting the fake machine id")
org.osbuild.ostree.passwd:    print("ostree " + " ".join(args), file=sys.stderr)
org.osbuild.dnf-automatic.config:        print(f"Error: DNF automatic configuration file '{dnf_automatic_config_path}' does not exist.")
org.osbuild.gcp.guest-agent.conf:        print(f"Creating new guest-agent configuration file at '{filepath}'.")
org.osbuild.rpmkeys.import:    print(inputs)
org.osbuild.rpmkeys.import:        print(f"imported key '{filename}'")
org.osbuild.uki:            print(f"skipping {name}")
org.osbuild.uki:    print(argv)
org.osbuild.ostree.encapsulate:    print("rpm-ostree " + " ".join(args), file=sys.stderr)
```

Reply:
Yeah I agree, this is quite a minor change. It makes me wonder if we should improve the way we print out information in the stages. Maybe by introducing proper logging ? (unrelated to the current PR however!)

Reply:
We should be ideally consistent in where we log (stdout vs. stderr), but I'm not sure if it makes sense to spend too much time on logging in stages.

Reply:
I'm all for proper logging and error reporting, but that seems like a larger task that doesn't belong in this PR :)

Reply:
Oh, definitely...